src/components/modal/confirm_modal.stories.tsx

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useState } from 'react';
+import type { Meta, StoryObj } from '@storybook/react';
+import { action } from '@storybook/addon-actions';
+
+import { moveStorybookControlsToCategory } from '../../../.storybook/utils';
+import { BUTTON_COLORS } from '../../themes/amsterdam/global_styling/mixins';
+import {
+  CANCEL_BUTTON,
+  CONFIRM_BUTTON,
+  EuiConfirmModal,
+  EuiConfirmModalProps,
+} from './confirm_modal';
+
+import { EuiButton } from '../button';
+
+const meta: Meta<EuiConfirmModalProps> = {
+  title: 'Layout/EuiConfirmModal',
+  component: EuiConfirmModal,
+  argTypes: {
+    // TODO: the `maxWidth` prop takes many different types (bool, string, number)
+    title: { control: { type: 'text' } },
+    confirmButtonText: { control: { type: 'text' } },
+    cancelButtonText: { control: { type: 'text' } },
+    defaultFocusedButton: {
+      control: { type: 'radio' },
+      options: [undefined, CONFIRM_BUTTON, CANCEL_BUTTON],
+    },
+    buttonColor: { control: { type: 'select' }, options: BUTTON_COLORS },
+    ...moveStorybookControlsToCategory(
+      ['role', 'maxWidth', 'initialFocus'],
+      'EuiModal props'
+    ),
+  },
+  args: {
+    buttonColor: 'primary',
+    role: 'alertdialog',
+    maxWidth: true,
+  },
+};
+
+export default meta;
+type Story = StoryObj<EuiConfirmModalProps>;
+
+const onCancel = action('onCancel');
+const onConfirm = action('onConfirm');
+
+const StatefulConfirmModal = (props: EuiConfirmModalProps) => {
+  const [isOpen, setIsOpen] = useState(true);
+  return (
+    <>
+      <EuiButton size="s" onClick={() => setIsOpen(!isOpen)}>
+        Toggle ConfirmModal
+      </EuiButton>
+      {isOpen && (
+        <EuiConfirmModal
+          {...props}
+          onConfirm={(...args) => {
+            setIsOpen(false);
+            onConfirm(...args);
+          }}
+          onCancel={(...args) => {
+            setIsOpen(false);
+            onCancel(...args);
+          }}
+        />
+      )}
+    </>
+  );
+};
+
+export const Playground: Story = {
+  args: {
+    title: 'Confirm modal title',
+    children: 'Confirm modal content',
+    confirmButtonText: 'Confirm',
```

Comment:
We should have a cancelButtonText example in the playground as well
```suggestion
    cancelButtonText: 'Cancel',
    confirmButtonText: 'Confirm',
```

Reply:
Thanks for the catch!